Xamarin.Forms.Controls/CoreGallery.cs

```diff
@@ -573,7 +581,19 @@ public CoreRootPage(Page rootPage, NavigationBehavior navigationBehavior = Navig
 				Command = new Command(async () =>
 				{
 					if (!string.IsNullOrEmpty(searchBar.Text))
-						await corePageView.PushPage(searchBar.Text);
+					{
+						if(!(await corePageView.PushPage(searchBar.Text)))
```

Comment:
All the gallery code added is to enable navigating to store shell from the root page of the gallery